### PR TITLE
Add Barricades and other misc improvements

### PR DESCRIPTION
**Pull request by thearst3rd**
* Add barricades which prevent enemies from progressing
* Fix animation speed not getting saved properly
* Add sound effect for charge card
* Add editorconfig for better github previewing

### CODE REVIEW

File: src/autoload/global.gd [hunk 1]
```diff
@@ -123,6 +125,7 @@ func _ready() -> void:
 			preload("res://src/cards/defense/defense_cards/oil_1.tres"),
 			preload("res://src/cards/defense/defense_cards/walls_1.tres"),
 			preload("res://src/cards/defense/defense_cards/mobilize_1.tres"),
+			preload("res://src/cards/defense/defense_cards/barricade_1.tres"),
```
Comment (admoore0) [inline src/autoload/global.gd:128]: Do you want to add higher ranks with more health?

File: src/cards/actions/create_barricade_unit.gd [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+		return false
+	return true
+
+
+func perform_action(grid_pos: Vector2i, is_enemy: bool) -> void:
+	var new_unit: BarricadeUnit = preload("res://src/units/barricade_unit.tscn").instantiate()
+	for unit in barricade_units.get_children():
+		if unit.grid_position == grid_pos:
+			barricade_units.remove_child(unit)
+			unit.queue_free()
```
Comment (admoore0) [inline src/cards/actions/create_barricade_unit.gd]: Remove

File: src/cards/actions/oil.gd [hunk 3]
```diff
@@ -18,16 +18,19 @@ func perform_action(grid_pos: Vector2i, is_enemy: bool) -> void:
...
-			unit.health -= damage
+			unit.health = maxi(unit.health - damage, 0)
 			unit.play_damage_sound()
 			unit.update_health_bar()
 
 	await game.wait_for_timer(Global.animation_speed)
 
 	for unit in melee_units.get_children():
 		if unit.health <= 0:
+			melee_units.remove_child(unit)
```
Comment (admoore0) [inline src/cards/actions/oil.gd:33]: How did we not catch this before?

File: src/states/game/game.gd [hunk 4]
```diff
@@ -465,51 +476,73 @@ func offensive_action_sweep() -> void:
 				unit.update_position()
 				steps_left -= 1
 				await wait_for_timer(Global.animation_speed)
+			elif unit.grid_position.x < 7: # We're not at the end, we got blocked
+				var blocking_unit := get_unit(unit.grid_position + Vector2i.RIGHT)
+				assert(blocking_unit, "Blocking unit is null?")
+				if $Units/Barricade.is_ancestor_of(blocking_unit):
```
Comment (admoore0) [inline src/states/game/game.gd]: Does `if blocking_unit is BarricadeUnit` not work here?

**Review by admoore0 — CHANGES_REQUESTED** [review]

File: src/cards/actions/oil.gd (hunk 3 above)
Comment (thearst3rd) [reply src/cards/actions/oil.gd:33]: idk fam skull emoji

**Review by admoore0 — APPROVED** [review]

**Review by kdreese — APPROVED** [review]

**Pull request merged**